3-HT-OOP/src/Book.ts

```diff
@@ -0,0 +1,39 @@
+import { Page } from "./Page";
+import { Pages } from "./Pages";
+
+class Book implements IterableIterator<String> {
+  private pointer = 0;
+  constructor(public title:string,public author:string,public pages: Pages) {}
+  public next(): IteratorResult<String> {
```

Comment:
Can you have a mixin which will implement the next instead of each class having to implement `next` method